.github/workflows/build-and-deploy-prod.yml

```
@@ -0,0 +1,44 @@
+  push:
+    branches:
+      - master
+
+jobs:
+  build_and_deploy_job:
+    runs-on: ubuntu-latest
+    env:
+      CI: false
+    name: Build and Deploy Job
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)